app/src/main/java/nu/yona/app/ui/message/NotificationFragment.java

```diff
@@ -75,137 +76,46 @@ public void onFriendsItemClick(View view)
 		{
 			if (view.getTag() instanceof YonaMessage)
 			{
-				YonaMessage yonaMessage = (YonaMessage) view.getTag();
-				Intent mMessageIntent = null;
-				if (yonaMessage.getNotificationMessageEnum() == NotificationMessageEnum.SYSTEM_MESSAGE)
+				yonaMessageClicked = (YonaMessage) view.getTag();
+				switch (yonaMessageClicked.getNotificationMessageEnum().getNotificationEnum())
 				{
-					mMessageIntent = new Intent(IntentEnum.ACTION_ADMIN_MESSAGE_DETAIL.getActionString());
-					mMessageIntent.putExtra(AppConstant.ADMIN_MESSAGE, yonaMessage);
-					mMessageIntent.putExtra(AppConstant.YONA_THEME_OBJ, new YonaHeaderTheme(false, null, null, 0, 0, null, R.color.grape, R.drawable.triangle_shadow_grape));
-					YonaAnalytics.createTapEventWithCategory(AnalyticsConstant.NOTIFICATION, AnalyticsConstant.ADMIN_MESSAGE_SCREEN);
+					case SYSTEMMESSAGE:
+						handleSystemMessageClick();
+						break;
+					case BUDDYCONNECTREQUESTMESSAGE:
+						handleBuddyConnectRequestMessageClick();
+						break;
+					case BUDDYCONNECTRESPONSEMESSAGE:
+						handleBuddyConnectResponseMessageClick();
+						break;
+					case ACTIVITYCOMMENTMESSAGE:
+						handleActivityCommentMessageClick();
+						break;
+					case GOALCONFLICTMESSAGE:
+						handleGoalConflictMessageClick();
+						break;
+					case GOALCHANGEMESSAGE:
+						handleGoalChangeMessageClick();
+						break;
+					case BUDDYINFOCHANGEMESSAGE:
+						handleBuddyInfoChangeMessageClick();
+						break;
+					default:
+						messageIntentClicked = null;
```

Comment:
Why would this need to be a member variable? It looks like another variable that should be passed over the stack instead, this time as return value of the various ``handleXxx`` methods, so you can pass it as input to ``replaceFragment`` below.